equinix-openapi-metal/src/main/java/com/equinix/workflow/MetalGatewayOperator.java

```diff
@@ -0,0 +1,125 @@
+import com.equinix.openapi.metal.v1.api.VlansApi;
+import com.equinix.openapi.metal.v1.model.CreateMetalGatewayRequest;
+import com.equinix.openapi.metal.v1.model.FindMetalGatewayById200Response;
+import com.equinix.openapi.metal.v1.model.Href;
+import com.equinix.openapi.metal.v1.model.MetalGateway;
+import com.equinix.openapi.metal.v1.model.MetalGateway.StateEnum;
+import com.equinix.openapi.metal.v1.model.MetalGatewayInput;
+import com.equinix.openapi.metal.v1.model.VirtualNetwork;
+import com.equinix.openapi.metal.v1.model.VirtualNetworkCreateInput;
+
+public class MetalGatewayOperator {
+    private MetalGatewaysApi metalGatewaysApi;
+    private VlansApi vlansApi;
+    private final List<StateEnum> targetState= Arrays.asList(StateEnum.ACTIVE, StateEnum.READY);
+
+    public MetalGatewayOperator(ApiClient apiClient) {
+        this.metalGatewaysApi = new MetalGatewaysApi(apiClient);
+        this.vlansApi = new VlansApi(apiClient);
+    }
+
+    // Returns the metal gateway queried by its id
+    public MetalGateway getMetalGateway(UUID metalGatewayId, List<String> include, List<String> exclude) throws ApiException {
+        FindMetalGatewayById200Response response = metalGatewaysApi.findMetalGatewayById(metalGatewayId, include, exclude);
+        return response.getMetalGateway();
+    }
+
+    // Returns the vlan queried by its id
+    public VirtualNetwork getVlan(UUID vlanId) throws ApiException {
+        return vlansApi.getVirtualNetwork(vlanId);
+    }
+
+    // Returns the vlan associated with the metal gateway queried by its id
+    public VirtualNetwork getVlanOfMetalGateway(UUID metalGatewayId) throws ApiException {
+        MetalGateway metalGateway = getMetalGateway(metalGatewayId, Collections.singletonList("virtual_network"), null);
+        return metalGateway.getVirtualNetwork();
+    }
+
+    // Returns a vlan in a project metro by its vxlan identifier
+    public VirtualNetwork getVlanByVxlanInProjectMetro(UUID projectId, String metro, Integer vxlan) throws ApiException {
+        VirtualNetwork vlan = null;
+        List<VirtualNetwork> vlanList = vlansApi.findVirtualNetworks(projectId, null, null, null, metro).getVirtualNetworks();
+        if (vlanList!=null && !vlanList.isEmpty()) {
+            Optional<VirtualNetwork> requiredVlan = vlanList.stream()
+                    .filter(vlanObj -> Objects.equals(vlanObj.getVxlan(), vxlan))
+                    .findFirst();
+            if (requiredVlan.isPresent())
+                vlan = requiredVlan.get();
+        }
+        return vlan;
+    }
+
+    // Create a vlan
+    public VirtualNetwork createVlan(UUID projectId, String metro, Integer vxlan) throws ApiException {
+        VirtualNetworkCreateInput vlanInput=new VirtualNetworkCreateInput()
+                .metro(metro)
+                .projectId(projectId)
+                .vxlan(vxlan);
+        return vlansApi.createVirtualNetwork(projectId, vlanInput);
+    }
+
+    // Metal Gateway creation workflow
```

Comment:
This javadoc comment should be revised to highlight that this only implements Gateway creation using private IP blocks.

We may want to introduce other operator helpers for creating IP or VRF reservation-based gateways, so this comment should not leave ambiguity about what this method does.

Reply:
Addressed in https://github.com/equinix-labs/metal-java/commit/4468cf4ebc426b7a3f5dd838ae9b878182ee1e81 and https://github.com/equinix-labs/metal-java/commit/6f781154eae8447cc8c1578d94f785713e8f35c4